Runtime/DisguiseRenderStream.cs

```diff
@@ -191,68 +191,78 @@ protected void CreateStreams()
             const int cullUIOnly = ~(1 << 5);
             
             ScratchTexture2DManager.Instance.Clear();
 
-            for (int i = 0; i < Streams.Length; ++i)
-            {        
-                StreamDescription stream = Streams[i];
-                Camera channelCamera = DisguiseRenderStream.GetChannelCamera(stream.channel);
+            for (int i = 0; i < m_Streams.Length; ++i)
+            {
+                StreamDescription stream = streams[i];
+                m_Streams[i].description = stream;
+                
+                Camera channelCamera = GetChannelCamera(stream.channel);
+                GameObject cameraObject;
                 if (channelCamera)
                 {
-                    m_Cameras[i] = UnityEngine.Object.Instantiate(channelCamera.gameObject, channelCamera.gameObject.transform);
-                    m_Cameras[i].name = stream.name;
+                    cameraObject = UnityEngine.Object.Instantiate(channelCamera.gameObject, channelCamera.gameObject.transform);
+                    cameraObject.name = stream.name;
                 }
                 else if (Camera.main)
                 {
-                    m_Cameras[i] = UnityEngine.Object.Instantiate(Camera.main.gameObject, Camera.main.gameObject.transform);
-                    m_Cameras[i].name = stream.name;
+                    cameraObject = UnityEngine.Object.Instantiate(Camera.main.gameObject, Camera.main.gameObject.transform);
+                    cameraObject.name = stream.name;
                 }
                 else
                 {
-                    m_Cameras[i] = new GameObject(stream.name);
-                    m_Cameras[i].AddComponent<Camera>();
+                    cameraObject = new GameObject(stream.name);
+                    cameraObject.AddComponent<Camera>();
                 }
 
-                m_Cameras[i].transform.localPosition = Vector3.zero;
-                m_Cameras[i].transform.localRotation = Quaternion.identity;
+                cameraObject.transform.localPosition = Vector3.zero;
+                cameraObject.transform.localRotation = Quaternion.identity;
             
-                GameObject cameraObject = m_Cameras[i];
                 Camera camera = cameraObject.GetComponent<Camera>();
+                m_Streams[i].camera = camera;
+                
                 camera.enabled = true; // ensure the camera component is enable
                 camera.cullingMask &= cullUIOnly; // cull the UI so RenderStream and other error messages don't render to RenderStream outputs
-                DisguiseCameraCapture capture = cameraObject.GetComponent(typeof(DisguiseCameraCapture)) as DisguiseCameraCapture;
-                if (capture == null)
-                    capture = cameraObject.AddComponent(typeof(DisguiseCameraCapture)) as DisguiseCameraCapture;
+            }
 
-                camera.enabled = true;
+            for (int i = 0; i < m_Streams.Length; ++i)
```

Comment:
Minor: Any specific reason this for loop is not combined with the previous one?

Reply:
There's [a loop](https://github.com/wenzhang-unity/RenderStream-Unity/pull/33/files#diff-8909d33fd94b66bb6de41b361ae848a93f7bc3e681aa105c880c2b893eeb4e7aR35) in `DisguiseCameraCapture.Awake()` that relies on the Streams list being partially filled :(

It would be more correct to fix the initialization, it shouldn't do a lookup but I don't see an obvious way to do that.